cli/js/40_lint.js

```diff
@@ -122,55 +149,46 @@ function getNode(ctx, offset) {
  * @returns {number}
  */
 function findPropOffset(buf, offset, search) {
-  // type + parentId + SpanLo + SpanHi
-  offset += 1 + 4 + 4 + 4;
-
-  const propCount = buf[offset];
+  const count = buf[offset];
   offset += 1;
 
-  for (let i = 0; i < propCount; i++) {
+  for (let i = 0; i < count; i++) {
     const maybe = offset;
     const prop = buf[offset++];
     const kind = buf[offset++];
     if (prop === search) return maybe;
 
-    if (kind === PropFlags.Ref) {
-      offset += 4;
-    } else if (kind === PropFlags.RefArr) {
+    if (kind === PropFlags.Obj) {
       const len = readU32(buf, offset);
-      offset += 4 + (len * 4);
-    } else if (kind === PropFlags.String) {
       offset += 4;
-    } else if (kind === PropFlags.Bool) {
-      offset++;
-    } else if (kind === PropFlags.Null || kind === PropFlags.Undefined) {
-      // No value
+      // prop + kind + value
+      offset += len * (1 + 1 + 4);
     } else {
-      offset++;
+      offset += 4;
     }
   }
 
   return -1;
 }
 
 const INTERNAL_CTX = Symbol("ctx");
-const INTERNAL_OFFSET = Symbol("offset");
+const INTERNAL_IDX = Symbol("offset");
 
 // This class is a facade for all materialized nodes. Instead of creating a
 // unique class per AST node, we have one class with getters for every
 // possible node property. This allows us to lazily materialize child node
 // only when they are needed.
-class Node {
+class FacadeNode {
```

Comment:
Do we really have to name it `FacadeNode`?

Reply:
The rename is needed because there is a `Node` type import already and even though it's merely a type, TS complains about a duplicated identifier. I suppose TS can only distinguish between type and class implementation in `.ts/tsx` files.